### PR TITLE
Small improvements to `workspace/symbol`

### PR DESCRIPTION
**Pull request by Duhemm**
- Inspect only source trees
 - Exclude local symbols

### CODE REVIEW

**Comment by Duhemm** [comment]
Rebased

File: language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala [hunk 1]
```diff
@@ -466,12 +471,14 @@ class DottyLanguageServer extends LanguageServer
 
   override def symbol(params: WorkspaceSymbolParams) = computeAsync { cancelToken =>
     val query = params.getQuery
+    def predicate(implicit ctx: Context): NameTree => Boolean =
+      tree => tree.symbol.exists && !tree.symbol.isLocal && tree.name.toString.contains(query)
```
Comment (smarter) [inline language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala]: Shouldn't symbol and documentSymbol have similar predicates ? I.e., should we also exclude primary constructors and synthetic symbols here ?

Comment (Duhemm) [reply language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala]: Yes, that makes a lot of sense. I've removed checks that were redundant with those happening in `namedTrees`.

I'm also thinking that `namedTrees` should be the one excluding primary constructors. I can't think of situations where they should be included, but I may very well be missing something. What do you think?

**Comment by smarter** [comment]
Agreed, as long as go to definition on a new call still goes to the correct place

File: language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala [hunk 2]
```diff
@@ -457,10 +457,7 @@ class DottyLanguageServer extends LanguageServer
     implicit val ctx = driver.currentCtx
 
     val uriTrees = driver.openedTrees(uri)
-    val predicate = (tree: NameTree) => {
-      val sym = tree.symbol
-      !sym.isLocal && !sym.isPrimaryConstructor
-    }
+    val predicate = (tree: NameTree) => !tree.symbol.isLocal
```
Comment (smarter) [inline language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala]: Maybe we could add `local` as a flag in `Include`, then we wouldn't need a custom predicate here (and as a bonus we could make the implementation smarter and not traverse local trees at all when the `local` flag is not set.

File: compiler/src/dotty/tools/dotc/interactive/Interactive.scala [hunk 3]
```diff
@@ -317,40 +322,41 @@ object Interactive {
...
-   */
-  def namedTrees(trees: List[SourceTree], nameSubstring: String)
-   (implicit ctx: Context): List[SourceTree] = {
-    val predicate: NameTree => Boolean = _.name.toString.contains(nameSubstring)
-    namedTrees(trees, Include.empty, predicate)
-  }
-
   /** Find named trees with a non-empty position satisfying `treePredicate` in `trees`.
    *
    *  @param includeReferences  If true, include references and not just definitions
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/interactive/Interactive.scala]: The doc is out of date: this parameter doesn't exist anymore.

**Pull request merged**